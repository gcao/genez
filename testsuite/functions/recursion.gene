#!/usr/bin/env gene run

# Test: Recursive functions
(fn factorial [n]
  (if (< n 2)
    1
    (* n (factorial (- n 1)))))

(print "(factorial 5) = " (factorial 5))
(print "(factorial 0) = " (factorial 0))

# Fibonacci
(fn fib [n]
  (if (< n 2)
    n
    (+ (fib (- n 1)) (fib (- n 2)))))

(print "(fib 10) = " (fib 10))