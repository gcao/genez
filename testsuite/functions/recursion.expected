(factorial 5) =  120
(factorial 0) =  1
(fib 10) =  55